### ISSUE TITLE
Emacs packages fail to build with error "Can't find define-package in"

### ISSUE THREAD

**Issue by lugeha** [nix-community/emacs-overlay#326]
```
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
no Makefile or custom buildPhase, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Parsing tar file... 
Parsing tar file...done

Error: error ("Can't find define-package in org-9.6.6/org-pkg.el")
  mapbacktrace(#f(compiled-function (evald func args flags) #<bytecode -0x20c71a12e510e41>))
  debug-early-backtrace()
  debug-early(error (error "Can't find define-package in org-9.6.6/org-pkg.el"))
  error("Can't find define-package in %s" "org-9.6.6/org-pkg.el")
  package-tar-file-info()
  (if (derived-mode-p 'tar-mode) (package-tar-file-info) (package-buffer-info))
  (let ((pkg-desc (if (derived-mode-p 'tar-mode) (package-tar-file-info) (package-buffer-info)))) (package-unpack pkg-desc) pkg-desc)
  elpa2nix-install-from-buffer()
  (progn (if is-tar (insert-file-contents-literally file) (insert-file-contents file)) (if is-tar (progn (tar-mode))) (elpa2nix-install-from-buffer))
  (unwind-protect (progn (if is-tar (insert-file-contents-literally file) (insert-file-contents file)) (if is-tar (progn (tar-mode))) (elpa2nix-install-from-buffer)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))
  (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (if is-tar (insert-file-contents-literally file) (insert-file-contents file)) (if is-tar (progn (tar-mode))) (elpa2nix-install-from-buffer)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))
  (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (if is-tar (insert-file-contents-literally file) (insert-file-contents file)) (if is-tar (progn (tar-mode))) (elpa2nix-install-from-buffer)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))))
  (let ((is-tar (string-match "\\.tar\\'" file))) (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (if is-tar (insert-file-contents-literally file) (insert-file-contents file)) (if is-tar (progn (tar-mode))) (elpa2nix-install-from-buffer)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))))
  elpa2nix-install-file("/nix/store/2j84jbz9pw25hs4jz3m3r605agd4a08z-org-9.6.6.tar")
  (progn (setq package-user-dir elpa) (elpa2nix-install-file archive))
  (let ((archive x0) (elpa x2)) (progn (setq package-user-dir elpa) (elpa2nix-install-file archive)))
  (if (null x3) (let ((archive x0) (elpa x2)) (progn (setq package-user-dir elpa) (elpa2nix-install-file archive))))
  (let* ((x2 (car-safe x1)) (x3 (cdr-safe x1))) (if (null x3) (let ((archive x0) (elpa x2)) (progn (setq package-user-dir elpa) (elpa2nix-install-file archive)))))
  (if (consp x1) (let* ((x2 (car-safe x1)) (x3 (cdr-safe x1))) (if (null x3) (let ((archive x0) (elpa x2)) (progn (setq package-user-dir elpa) (elpa2nix-install-file archive))))))
  (let* ((x0 (car-safe command-line-args-left)) (x1 (cdr-safe command-line-args-left))) (if (consp x1) (let* ((x2 (car-safe x1)) (x3 (cdr-safe x1))) (if (null x3) (let ((archive x0) (elpa x2)) (progn (setq package-user-dir elpa) (elpa2nix-install-file archive)))))))
  (if (consp command-line-args-left) (let* ((x0 (car-safe command-line-args-left)) (x1 (cdr-safe command-line-args-left))) (if (consp x1) (let* ((x2 (car-safe x1)) (x3 (cdr-safe x1))) (if (null x3) (let ((archive x0) (elpa x2)) (progn (setq package-user-dir elpa) (elpa2nix-install-file archive))))))))
  elpa2nix-install-package()
  command-line-1(("-l" "/nix/store/4jj63z4v1xp13rh2md053dccq920hd45-elpa2nix.el" "-f" "elpa2nix-install-package" "/nix/store/2j84jbz9pw25hs4jz3m3r605agd4a08z-org-9.6.6.tar" "/nix/store/4y0qbsnmiaymffiq5m4z6iss73ddz82r-emacs-org-9.6.6/share/emacs/site-lisp/elpa"))
  command-line()
  normal-top-level()
Can't find define-package in org-9.6.6/org-pkg.el
```

I'm not sure where the problem was introduced, but if I roll back to **github:nix-community/emacs-overlay/338da822b1506c26c18e6fced9527a8de4f08665** it goes away (this was my last known working flake lock)

**Comment by kira-bruneau** [comment]
Hi @lugeha! I'm running into the same issue, and I'm just curious why you closed this.

**Comment by leungbk** [comment]
> Hi @lugeha! I'm running into the same issue, and I'm just curious why you closed this.

Do you still encounter this with the latest master? I am on 23dfd9c7e508ab17b3171c8c37551479ce845c2c and see no issues. I tried to work around this in https://github.com/nix-community/emacs-overlay/commit/e7cd4e4053af9872840f04e7a36c5f11aa81648b

**Comment by kira-bruneau** [comment]
I was seeing these errors on 8b10f751e48ae0e48b3f037a352c7447dc7d1afe, but I'll try again just to make sure (338da822b1506c26c18e6fced9527a8de4f08665 worked for me though)

**Comment by leungbk** [comment]
I can't reproduce any issue with https://github.com/nix-community/emacs-overlay/commit/8b10f751e48ae0e48b3f037a352c7447dc7d1afe, if you can please provide a repro.

**Comment by kira-bruneau** [comment]
Oh yep still seeing the same errors with 8b10f751e48ae0e48b3f037a352c7447dc7d1afe - Oh maybe it's a problem with the version of nixpkgs I'm using? I have my flake setup to follow my flake's nixpkgs, and it's set at `aeb75dba965e790de427b73315d5addf91a54955`

**Comment by kira-bruneau** [comment]
Oh wait no, that's not it it still breaks when I leave it unpinned 😕

**Comment by leungbk** [comment]
What Emacs packages are you installing? I'd like to try reproducing this myself. The problem I sought to work around was an upstream thing that seems to impact all natively-compiled packages (https://lists.gnu.org/archive/html/bug-gnu-emacs/2023-05/msg01512.html), but I don't install that many Emacs packages with Nix.

**Comment by kira-bruneau** [comment]
Oh, here's the full set: https://gitlab.com/kira-bruneau/home-config/-/blob/main/package/emacs/package/default.nix#L88-164 - I'm just rebuilding right now to get the set that fail for me.

**Comment by kira-bruneau** [comment]
Oh sorry it took so long, ssh-ng was hanging on one of my builders for some reason:
error: builder for '/nix/store/1dmmawsh29n1m46a0iikyynf34mc5bd4-emacs-adaptive-wrap-0.8.drv' failed with exit code 255
error: builder for '/nix/store/9pzlfhsvssvhnw444hxj30q5k87izrfy-emacs-apheleia-20230527.1755.drv' failed with exit code 255
error: builder for '/nix/store/mp8ldmxlw2xa053h17zwm9hf9dfwfkhv-emacs-async-20230528.622.drv' failed with exit code 255
error: builder for '/nix/store/3hzhrpnjbkmfqk94p0w0nh4yj3qrl9zd-emacs-avy-20230420.404.drv' failed with exit code 255
error: builder for '/nix/store/yi0xakhfgp0ysbcy3qq4q5qsj1hmnprk-emacs-beacon-20220730.100.drv' failed with exit code 255
error: builder for '/nix/store/hcgycc6318m6hkqsa11glxfagxjbcr87-emacs-buffer-move-20220512.755.drv' failed with exit code 255
error: builder for '/nix/store/ri2i3d56ds9sl65v7a4m7msxxx4mbk7i-emacs-dockerfile-mode-20220822.2021.drv' failed with exit code 255
error: builder for '/nix/store/wqg2vivrycha01by8kdv6jj5mflh3xn3-emacs-diminish-20220909.847.drv' failed with exit code 255
error: builder for '/nix/store/q66x07wglql4bpwiff6ms86jii4dhxfs-emacs-drag-stuff-20161108.749.drv' failed with exit code 255
error: builder for '/nix/store/n9id2frfbzgas45b8gp80jwr7h62csxf-emacs-compat-29.1.4.1.drv' failed with exit code 255
error: builder for '/nix/store/ajajr4gss0hbn1a78j1m8z6bfxr90smr-emacs-doom-themes-20230221.610.drv' failed with exit code 255
error: builder for '/nix/store/hf5gfrf0ysjljsp02n4a5nb012r6k6ip-emacs-annalist-20190929.207.drv' failed with exit code 255
error: builder for '/nix/store/dq2jlfbhb6lgz612wsbl9jihwjsiazw5-emacs-clojure-mode-20230314.758.drv' failed with exit code 255
error: builder for '/nix/store/mdy9bcn4hmkarr4fyhpsyyadpbsim0j3-emacs-company-20230416.2034.drv' failed with exit code 255
error: builder for '/nix/store/xwlldh8pnq0fy9aaak603qqk5v4hlgsw-emacs-eldoc-1.14.0.drv' failed with exit code 255
error: builder for '/nix/store/2xx13dp3d34p5y5llbqvfy34hcaz3q90-emacs-dtrt-indent-20230302.2151.drv' failed with exit code 255
error: builder for '/nix/store/bjs3pj24ggr89im83c5lv5d8npgva3ns-emacs-cmake-mode-20230422.828.drv' failed with exit code 255
error: builder for '/nix/store/633kfzm8mga8lm33816g1l0q7lswc197-emacs-epl-20180205.2049.drv' failed with exit code 255
error: builder for '/nix/store/xs3cj3c442hpbhkrbbzn28w6q1ig8wah-emacs-flx-20221213.542.drv' failed with exit code 255
error: builder for '/nix/store/zbvg6hpd4j0yb5v2snn25nw2mlr07hrd-emacs-dash-20230502.1324.drv' failed with exit code 255
error: builder for '/nix/store/aaq9sk05bnb6cjil2fmg9bhh0wjd9i5q-emacs-gcmh-20201116.2251.drv' failed with exit code 255
error: builder for '/nix/store/065is51qvgk0fppdyr78jszp2js1p4lj-emacs-go-mode-20221228.1706.drv' failed with exit code 255
error: builder for '/nix/store/zpfjadaw7vnfi7z387a02yndgg61b1bx-emacs-goto-chg-20220107.1733.drv' failed with exit code 255
error: builder for '/nix/store/xmfr9ag4bdbq1av74czdrwg28rzrlh7r-emacs-graphql-mode-20230411.1943.drv' failed with exit code 255
error: builder for '/nix/store/z6xlw1mp6xgziv4l47vkngf947z9alkc-emacs-js2-mode-20230509.2244.drv' failed with exit code 255
error: builder for '/nix/store/980hkb4a3dss82wrddydnbw763g2b2gr-emacs-ivy-20230410.1815.drv' failed with exit code 255
error: builder for '/nix/store/c33a1xymj4b4cfcma1lr08ssg63qwmz1-emacs-json-snatcher-20200916.1717.drv' failed with exit code 255
error: builder for '/nix/store/433lidpir1ir0j38zip222rp3v936b8s-emacs-let-alist-1.0.6.drv' failed with exit code 255
error: builder for '/nix/store/fn62ivyq8gs3ipxsbkw3y29cxnqq07pz-emacs-know-your-http-well-20160208.2305.drv' failed with exit code 255
error: builder for '/nix/store/d53rpz1kxr03kf0l5qivw5xya43f868k-emacs-latex-preview-pane-20181008.1822.drv' failed with exit code 255
error: builder for '/nix/store/znksiig80qr74mjnbqwvcfp6h9lc6zk8-emacs-map-3.3.1.drv' failed with exit code 255
error: builder for '/nix/store/97h5xxifbzainy51v9gvrk45r6cmm7pb-emacs-loop-20160813.1407.drv' failed with exit code 255
error: builder for '/nix/store/fkikw82xvi46g952k7g1zad4hdvdc6yr-emacs-lv-20200507.1518.drv' failed with exit code 255
error: builder for '/nix/store/7q15iv0a1y3mwxwn63j6x81lqlii306b-emacs-macrostep-20230518.1911.drv' failed with exit code 255
error: builder for '/nix/store/29xh97s0ban67d4zz4j8mn6d9221xidh-emacs-lua-mode-20221218.605.drv' failed with exit code 255
error: builder for '/nix/store/fqj6bnzbr4db4r876nv3hrdgk2h5nmfn-emacs-markdown-mode-20230522.147.drv' failed with exit code 255
error: builder for '/nix/store/qg6azhbq26p0svk6mzy54m15ay5jnzrq-emacs-mermaid-mode-20230517.1527.drv' failed with exit code 255
error: builder for '/nix/store/g5asx8f1w0j2rlhaawwf0yav496m0jfq-emacs-multi-term-20200514.428.drv' failed with exit code 255
error: builder for '/nix/store/23fjgdp7dgadyk1gjwahldcjr5sn0sfw-emacs-multiple-cursors-20230309.1204.drv' failed with exit code 255
error: builder for '/nix/store/66k387rxnywxv9hz16sf1v6z1gjlj5r6-emacs-nadvice-0.4.drv' failed with exit code 255
error: builder for '/nix/store/nr3cd0qgw77qw30a3kyd5cki6jfg15zk-emacs-nameless-20230112.1259.drv' failed with exit code 255
error: builder for '/nix/store/a4mdkw6iaznbsljyjg3p453mm32v0s81-emacs-page-break-lines-20210104.2224.drv' failed with exit code 255
error: builder for '/nix/store/81z15nxx30hqdiqd7pn42vyirh2d6yw3-emacs-parseclj-20220422.936.drv' failed with exit code 255
error: builder for '/nix/store/xq6gvkd30m771bjq8zmb3lqim2mfb6bn-emacs-pfuture-20220913.1401.drv' failed with exit code 255
error: builder for '/nix/store/4bz6zbq0793k11paf7xd41baqg4ix5mp-emacs-pkgbuild-mode-20220802.1951.drv' failed with exit code 255
error: builder for '/nix/store/fs5s35fwpv6zrlx7mmq2wxkblb2dlhc5-emacs-php-mode-20230521.1700.drv' failed with exit code 255
error: builder for '/nix/store/16c96k19wcixxbcjg4m12yskpwl9n5bv-emacs-posframe-20230423.927.drv' failed with exit code 255
error: builder for '/nix/store/c42kin020bmpaccbnb44yw6mpd5zdnk7-emacs-powershell-20220805.1712.drv' failed with exit code 255
error: builder for '/nix/store/3k6182bl1vq9ffqnm3im339hwyb3yj4v-emacs-powerline-20221110.1956.drv' failed with exit code 255
error: builder for '/nix/store/y47hg37dc411xplva78bn5q1n9xbf1qr-emacs-presentation-20180427.224.drv' failed with exit code 255
error: builder for '/nix/store/im7483v0p10k9i7nd3az0m50z9hj6kks-emacs-projectile-20230317.1101.drv' failed with exit code 255
error: builder for '/nix/store/jqaah1zxfv9dm1ggph1l7q8n20fin0h0-emacs-queue-0.2.drv' failed with exit code 255
error: builder for '/nix/store/201r21x1xr3s95a8lz4brranj7fhysdx-emacs-request-20230127.417.drv' failed with exit code 255
error: builder for '/nix/store/7s167q3xsc5k95zpvanf28sqnasgnw74-emacs-spinner-1.7.4.drv' failed with exit code 255
error: builder for '/nix/store/va3vsz8pvf22i91yl0h3vrnl1b24h3av-emacs-rainbow-delimiters-20210515.1254.drv' failed with exit code 255
error: builder for '/nix/store/8iwjl5q4q9wh5abg21cpyjplhlk6p8d2-emacs-restclient-20221203.1808.drv' failed with exit code 255
error: builder for '/nix/store/f6smdgh02fjcbvj38jdmjbqx09bpkjjh-emacs-rust-mode-20230112.27.drv' failed with exit code 255
error: builder for '/nix/store/0prj7affyyjnswbyx0qjs6a5g1b4vpsy-emacs-s-20220902.1511.drv' failed with exit code 255
error: builder for '/nix/store/krl2r7p7jpil38q4vwcndyv4a2qnbnyr-emacs-sesman-20210901.1134.drv' failed with exit code 255
error: builder for '/nix/store/46azyawbrqz1b381nncl73rjncyxy5x8-emacs-treepy-20230528.750.drv' failed with exit code 255
error: builder for '/nix/store/8x0gbz5khjrmav14ajr2bb39pa4miq9x-emacs-typescript-mode-20230116.1533.drv' failed with exit code 255
error: builder for '/nix/store/5941j9zxdij1w7dpll06399hs12xfj2l-emacs-vala-mode-20201218.2109.drv' failed with exit code 255
error: builder for '/nix/store/645zb606sqdxwcb30bi27zynkp2lsnbr-emacs-visual-regexp-20210502.2019.drv' failed with exit code 255
error: builder for '/nix/store/nr2bpqj0zc15wmcbcgjqdnw3lhvhrpjx-emacs-web-mode-20230329.601.drv' failed with exit code 255
error: builder for '/nix/store/v3hkb8rbilq7xjrrgf9m9ipa4gr8hg6q-emacs-wgrep-20230203.1214.drv' failed with exit code 255
error: builder for '/nix/store/4zqs6cxn4097hav9m1k0fqjpn803nn3w-emacs-xref-1.6.3.drv' failed with exit code 255
error: builder for '/nix/store/s3g6srrg8p66nxwijnnahls6z8xlxsv5-emacs-vlf-20191126.2250.drv' failed with exit code 255
error: builder for '/nix/store/hiz92jar6mkihjs2iav437zfmpsvx2zr-emacs-which-key-20230330.2121.drv' failed with exit code 255
error: builder for '/nix/store/83194kjjm0aws14ps5j761w2c08rbp04-emacs-whitespace-cleanup-mode-20210510.533.drv' failed with exit code 255
error: builder for '/nix/store/xs91m7w1bhiyl12braqm096f2py1ynkh-emacs-xterm-color-20230321.3.drv' failed with exit code 255
error: builder for '/nix/store/v7pqq2p847fk1sm46szfdjk0q272ra4q-emacs-yaml-mode-20230329.723.drv' failed with exit code 255
error: builder for '/nix/store/3jg6nxniq6nmwj6p25gdipm3bwlmly3a-emacs-yaml-20230312.250.drv' failed with exit code 255
error: builder for '/nix/store/wfp1nwzjzvzhapv42hgim13v7jldvwxp-emacs-emacsql-20230417.1448.drv' failed with exit code 255

**Comment by kira-bruneau** [comment]
And maybe this is more helpful to see how each of them are failing?:
```
emacs-adaptive-wrap> Error: error ("Can't find define-package in adaptive-wrap-0.8/adaptive-wrap-pkg.el")
emacs-apheleia> Error: error ("Can't find define-package in apheleia-20230527.1755/apheleia-pkg.el")
emacs-async> Error: error ("Can't find define-package in async-20230528.622/async-pkg.el")
emacs-buffer-move> Error: wrong-type-argument (package-desc nil)
emacs-avy> Error: wrong-type-argument (package-desc nil)
emacs-beacon> Error: wrong-type-argument (package-desc nil)
emacs-dockerfile-mode> Error: wrong-type-argument (package-desc nil)
emacs-diminish> Error: wrong-type-argument (package-desc nil)
emacs-doom-themes> Error: error ("Can't find define-package in doom-themes-20230221.610/doom-themes-pkg.el")
emacs-drag-stuff> Error: error ("Can't find define-package in drag-stuff-20161108.749/drag-stuff-pkg.el")
emacs-compat> Error: error ("Can't find define-package in compat-29.1.4.1/compat-pkg.el")
emacs-company> Error: error ("Can't find define-package in company-20230416.2034/company-pkg.el")
emacs-annalist> Error: error ("Can't find define-package in annalist-20190929.207/annalist-pkg.el")
emacs-clojure-mode> Error: wrong-type-argument (package-desc nil)
emacs-eldoc> Error: error ("Can't find define-package in eldoc-1.14.0/eldoc-pkg.el")
emacs-dtrt-indent> Error: error ("Can't find define-package in dtrt-indent-20230302.2151/dtrt-indent-pkg.el")
emacs-epl> Error: wrong-type-argument (package-desc nil)
emacs-cmake-mode> Error: wrong-type-argument (package-desc nil)
emacs-dash> Error: error ("Can't find define-package in dash-20230502.1324/dash-pkg.el")
emacs-flx> Error: wrong-type-argument (package-desc nil)
emacs-gcmh> Error: wrong-type-argument (package-desc nil)
emacs-go-mode> Error: wrong-type-argument (package-desc nil)
emacs-goto-chg> Error: wrong-type-argument (package-desc nil)
emacs-js2-mode> Error: error ("Can't find define-package in js2-mode-20230509.2244/js2-mode-pkg.el")
emacs-graphql-mode> Error: wrong-type-argument (package-desc nil)
emacs-ivy> Error: error ("Can't find define-package in ivy-20230410.1815/ivy-pkg.el")
emacs-json-snatcher> Error: wrong-type-argument (package-desc nil)
emacs-know-your-http-well> Error: error ("Can't find define-package in know-your-http-well-20160208.2305/know-your-http-well-pkg.el")
emacs-latex-preview-pane> Error: error ("Can't find define-package in latex-preview-pane-20181008.1822/latex-preview-pane-pkg.el")
emacs-let-alist> Error: wrong-type-argument (package-desc nil)
emacs-map> Error: error ("Can't find define-package in map-3.3.1/map-pkg.el")
emacs-loop> Error: wrong-type-argument (package-desc nil)
emacs-lua-mode> Error: wrong-type-argument (package-desc nil)
emacs-nadvice> Error: error ("Can't find define-package in nadvice-0.4/nadvice-pkg.el")
emacs-lv> Error: wrong-type-argument (package-desc nil)
emacs-macrostep> Error: error ("Can't find define-package in macrostep-20230518.1911/macrostep-pkg.el")
emacs-mermaid-mode> Error: wrong-type-argument (package-desc nil)
emacs-markdown-mode> Error: wrong-type-argument (package-desc nil)
emacs-multi-term> Error: wrong-type-argument (package-desc nil)
emacs-multiple-cursors> Error: error ("Can't find define-package in multiple-cursors-20230309.1204/multiple-cursors-pkg.el")
emacs-nameless> Error: wrong-type-argument (package-desc nil)
emacs-page-break-lines> Error: wrong-type-argument (package-desc nil)
emacs-parseclj> Error: error ("Can't find define-package in parseclj-20220422.936/parseclj-pkg.el")
emacs-pfuture> Error: wrong-type-argument (package-desc nil)
emacs-php-mode> Error: error ("Can't find define-package in php-mode-20230521.1700/php-mode-pkg.el")
emacs-pkgbuild-mode> Error: wrong-type-argument (package-desc nil)
emacs-posframe> Error: error ("Can't find define-package in posframe-20230423.927/posframe-pkg.el")
emacs-powerline> Error: error ("Can't find define-package in powerline-20221110.1956/powerline-pkg.el")
emacs-queue> Error: wrong-type-argument (package-desc nil)
emacs-presentation> Error: wrong-type-argument (package-desc nil)
emacs-powershell> Error: wrong-type-argument (package-desc nil)
emacs-projectile> Error: wrong-type-argument (package-desc nil)
emacs-rainbow-delimiters> Error: wrong-type-argument (package-desc nil)
emacs-request> Error: wrong-type-argument (package-desc nil)
emacs-spinner> Error: error ("Can't find define-package in spinner-1.7.4/spinner-pkg.el")
emacs-restclient> Error: wrong-type-argument (package-desc nil)
emacs-rust-mode> Error: error ("Can't find define-package in rust-mode-20230112.27/rust-mode-pkg.el")
emacs-sesman> Error: error ("Can't find define-package in sesman-20210901.1134/sesman-pkg.el")
emacs-s> Error: wrong-type-argument (package-desc nil)
emacs-treepy> Error: wrong-type-argument (package-desc nil)
emacs-typescript-mode> Error: error ("Can't find define-package in typescript-mode-20230116.1533/typescript-mode-pkg.el")
emacs-visual-regexp> Error: wrong-type-argument (package-desc nil)
emacs-xref> Error: error ("Can't find define-package in xref-1.6.3/xref-pkg.el")
emacs-vala-mode> Error: wrong-type-argument (package-desc nil)
emacs-vlf> Error: error ("Can't find define-package in vlf-20191126.2250/vlf-pkg.el")
emacs-wgrep> Error: wrong-type-argument (package-desc nil)
emacs-whitespace-cleanup-mode> Error: wrong-type-argument (package-desc nil)
emacs-web-mode> Error: wrong-type-argument (package-desc nil)
emacs-which-key> Error: wrong-type-argument (package-desc nil)
emacs-yaml> Error: wrong-type-argument (package-desc nil)
emacs-yaml-mode> Error: wrong-type-argument (package-desc nil)
emacs-xterm-color> Error: wrong-type-argument (package-desc nil)
emacs-emacsql> Error: error ("Can't find define-package in emacsql-20230417.1448/emacsql-pkg.el")
```

**Comment by leungbk** [comment]
Hmm, I still can't repro even when I try installing some of the packages you mention.

Multiple people have reported an issue that yields the same problems you mention (for example, https://lists.gnu.org/archive/html/bug-gnu-emacs/2023-05/msg01687.html). There has been an upstream attempt to work around the issue, so I have reverted the patch that I applied and updated the Emacs input accordingly. I'm not knowledgeable with the native-comp internals, so if this latest rev doesn't work for you, I'm afraid I can't be of much help.
